File: Array/Prime Sum.go
package main

import "fmt"

func main() {

	result := primesum(184)
	fmt.Println(result)
}

func primesum(A int) []int {
	ans := make([]int, 2)
	prime := make([]bool, A+1)
	for i := 0; i <= A; i++ {
		prime[i] = true
	}

	prime[0] = false
	prime[1] = false

	for i := 2; i <= A; i++ {

		if !prime[i] {
			continue
		}

		if i > A/i {
			break
		}

		for j := i * i; j <= A; j += i {
			prime[j] = false
		}
	}

	for i := 2; i <= A; i++ {
		if prime[i] && prime[A-i] {
			ans[0] = i
			ans[1] = A - i
			return ans
		}
	}

	return ans

}

File: Pythons Solutions/Prime Sum.py
class Solution:
    def primeSum(self, a):
        prime = [True]*a
        prime[0], prime[1] = False, False

        for i in range(2,a+1):
            if not prime[i]:
                continue
            if i > a/i:
                break

            j = i*i
            while j <a:
                prime[j] = False
                j += i
        
        ans = [0]*2
        
        for i in range(2, a+1):
            if prime[i] and prime[a-i]:
                ans[0] = i
                ans[1] = a-i
                break


        return ans

obj = Solution()
a = 6
ans = obj.primeSum(a)
print(ans)